src/main/java/org/testobject/appium/testng/TestObjectTestNGTestProvider.java

```diff
@@ -0,0 +1,52 @@
+package org.testobject.appium.testng;
+
+import io.appium.java_client.AppiumDriver;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.net.URL;
+
+public class TestObjectTestNGTestProvider {
+
+	private RemoteWebDriver remoteWebDriver;
+
+	private URL endpoint;
+
+	private boolean isLocalTest;
+
+	public static TestObjectTestNGTestProvider newInstance() {
+		return new TestObjectTestNGTestProvider();
```

Comment:
Rule **6** - https://github.com/testobject/best-practices - It increases encapsulation so you can do manipulations without break clients in the future.